staging/src/github.com/kubeedge/viaduct/pkg/fifo/msgfifo.go

```diff
@@ -43,3 +45,9 @@ func (f *MessageFifo) Get(msg *model.Message) error {
 	}
 	return nil
 }
+
+func (f *MessageFifo) Close() {
+	f.closeOnce.Do(func() {
+		close(f.fifo)
```

Comment:
Why do we only close once? Looks like the fifo will be recreated every time reconnect.

Reply:
Only same object close once, two different object will be closed again.

Reply:
Okey :), so for the same object, is it necessary to use once?

Reply:
yes, closing a closed channel will panic, so the channel should be closed once

Reply:
Make sense! And would you like to split the second commit to another pr?

Reply:
ok, I will split it to another pr.